### PR TITLE
FIX: Contact dolGetButtonAction

### PR DESCRIPTION
**Pull request by atm-steve**
Replace old butAction with dolGetButtonAction

### CODE REVIEW

File: htdocs/contact/card.php [hunk 1]
```diff
@@ -1480,36 +1480,44 @@ function init_check_no_email(input) {
...
 			if ($object->statut == 1 && $user->rights->societe->contact->creer) {
-				print '<a class="butActionDelete" href="'.$_SERVER['PHP_SELF'].'?action=disable&id='.$object->id.'&token='.newToken().'">'.$langs->trans("DisableUser").'</a>';
+
+				print dolGetButtonAction($langs->trans('DisableUser'), '', 'delete', $_SERVER['PHP_SELF'].'?action=disable&id='.$object->id.'&token='.newToken(), '', true, $params);
 			}
 
 			// Delete
 			if ($user->rights->societe->contact->supprimer) {
-				print '<a class="butActionDelete" href="'.$_SERVER['PHP_SELF'].'?id='.$object->id.'&action=delete&token='.newToken().''.($backtopage ? '&backtopage='.urlencode($backtopage) : '').'">'.$langs->trans('Delete').'</a>';
+				print dolGetButtonAction($langs->trans('Delete'), '', 'delete', $_SERVER['PHP_SELF'].'?id='.$object->id.'&action=delete&token='.newToken(), '', true, $params);
```
Comment (eldy) [inline htdocs/contact/card.php:1520]: It seem the parameter  $backtopage  is lost after this change. Can you check please ?

Comment (SteveATM) [reply htdocs/contact/card.php]: Ok all done. Thank you for this review.

File: htdocs/contact/card.php [hunk 2]
```diff
@@ -1481,35 +1481,51 @@ function init_check_no_email(input) {
...
-				print '<a class="butActionDelete" href="'.$_SERVER['PHP_SELF'].'?action=disable&id='.$object->id.'&token='.newToken().'">'.$langs->trans("DisableUser").'</a>';
+
+				print dolGetButtonAction($langs->trans('DisableUser'), '', 'delete', $_SERVER['PHP_SELF'].'?action=disable&id='.$object->id.'&token='.newToken(), '', true, $params);
 			}
 
 			// Delete
 			if ($user->rights->societe->contact->supprimer) {
-				print '<a class="butActionDelete" href="'.$_SERVER['PHP_SELF'].'?id='.$object->id.'&action=delete&token='.newToken().''.($backtopage ? '&backtopage='.urlencode($backtopage) : '').'">'.$langs->trans('Delete').'</a>';
+				$url = $_SERVER['PHP_SELF'].'?id='.$object->id.'&action=delete&token='.newToken();
+				$url.= '&backtopage='.urlencode(DOL_URL_ROOT . '/contact/list.php');
```
Comment (eldy) [inline htdocs/contact/card.php:1527]: If backtopage is not defined, we must not add it (see condition in old code).

File: htdocs/contact/card.php [hunk 3]
```diff
@@ -1481,35 +1481,55 @@ function init_check_no_email(input) {
...
+
+				print dolGetButtonAction($langs->trans('DisableUser'), '', 'delete', $_SERVER['PHP_SELF'].'?action=disable&id='.$object->id.'&token='.newToken(), '', true, $params);
 			}
 
 			// Delete
 			if ($user->rights->societe->contact->supprimer) {
-				print '<a class="butActionDelete" href="'.$_SERVER['PHP_SELF'].'?id='.$object->id.'&action=delete&token='.newToken().''.($backtopage ? '&backtopage='.urlencode($backtopage) : '').'">'.$langs->trans('Delete').'</a>';
+				$url = $_SERVER['PHP_SELF'].'?id='.$object->id.'&action=delete&token='.newToken();
+				if ($backtopage) {
+					$url.= '&backtopage='.urlencode(DOL_URL_ROOT . '/contact/list.php');
```
Comment (eldy) [inline htdocs/contact/card.php:1528]: If backtopage is defined, param must be set to value defined.

File: htdocs/contact/card.php [hunk 4]
```diff
@@ -1481,35 +1481,55 @@ function init_check_no_email(input) {
...
+				} else {
```
Comment (eldy) [inline htdocs/contact/card.php:1529]: else is useless

**Pull request closed without merging**